src/sidebar/helpers/annotation-segment.ts

```diff
@@ -22,6 +22,8 @@ export function annotationMatchesSegment(
   }
 
   return Boolean(
+    // nb. The URL comparison here assumes that both URLs are either absolute
+    // or relative to the same root.
```

Comment:
I added a note about this hazard but didn't change anything here. Aside from annotations creating while testing the `book_as_single_document` flag with earlier builds of the client, all selectors and segment URLs will be absolute. For those annotations, the CFI-based comparison here will be an adequate fallback.